diff --git a/KeyConfig.cpp b/KeyConfig.cpp
index 2aadbd8..da4cb39 100644
--- a/KeyConfig.cpp
+++ b/KeyConfig.cpp
@@ -100,6 +100,9 @@ map<int, int> KeyConfig::buildDefaultKeymap()
 
     keymap['1'] = ACTION_DECREASE_SPEED;
     keymap['2'] = ACTION_INCREASE_SPEED;
+    keymap['8'] = ACTION_DECREASE_SPEED_MICRO;
+    keymap['9'] = ACTION_NORMAL_SPEED;
+    keymap['0'] = ACTION_INCREASE_SPEED_MICRO;
     keymap['<'] = ACTION_REWIND;
     keymap[','] = ACTION_REWIND;
     keymap['>'] = ACTION_FAST_FORWARD;
diff --git a/KeyConfig.h b/KeyConfig.h
index 7de06c7..53b52e0 100644
--- a/KeyConfig.h
+++ b/KeyConfig.h
@@ -43,6 +43,9 @@ class KeyConfig
         ACTION_CROP_VIDEO = 34,
         ACTION_PAUSE = 35,
         ACTION_PLAY = 36,
+        ACTION_DECREASE_SPEED_MICRO = 37,
+        ACTION_NORMAL_SPEED = 38,
+        ACTION_INCREASE_SPEED_MICRO = 39
     };
 
     #define KEY_LEFT 0x5b44
diff --git a/omxplayer.cpp b/omxplayer.cpp
index bde4536..3dc87ad 100644
--- a/omxplayer.cpp
+++ b/omxplayer.cpp
@@ -96,6 +96,7 @@ bool              m_Pause               = false;
 OMXReader         m_omx_reader;
 int               m_audio_index_use     = 0;
 OMXClock          *m_av_clock           = NULL;
+int               m_fpp_speed           = 1000;
 OMXControl        m_omxcontrol;
 Keyboard          *m_keyboard           = NULL;
 OMXAudioConfig    m_config_audio;
@@ -1204,6 +1205,26 @@ int main(int argc, char *argv[])
         m_tv_show_info = !m_tv_show_info;
         vc_tv_show_info(m_tv_show_info);
         break;
+      case KeyConfig::ACTION_DECREASE_SPEED_MICRO:
+        if (m_fpp_speed >= 33)
+          m_fpp_speed -= 33;
+        SetSpeed(m_fpp_speed);
+        m_Pause = false;
+        break;
+      case KeyConfig::ACTION_NORMAL_SPEED:
+        if (m_fpp_speed != 1000)
+        {
+          m_fpp_speed = 1000;
+          SetSpeed(m_fpp_speed);
+          m_Pause = false;
+        }
+        break;
+      case KeyConfig::ACTION_INCREASE_SPEED_MICRO:
+        if (m_fpp_speed <= 1467)
+          m_fpp_speed += 33;
+        SetSpeed(m_fpp_speed);
+        m_Pause = false;
+        break;
       case KeyConfig::ACTION_DECREASE_SPEED:
         if (playspeed_current < playspeed_slow_min || playspeed_current > playspeed_slow_max)
           playspeed_current = playspeed_slow_max-1;
@@ -1636,7 +1657,7 @@ int main(int argc, char *argv[])
       if(m_stats)
       {
         static int count;
-        if ((count++ & 7) == 0)
+//        if ((count++ & 7) == 0)
            printf("M:%8.0f V:%6.2fs %6dk/%6dk A:%6.2f %6.02fs/%6.02fs Cv:%6dk Ca:%6dk                            \r", stamp,
                video_fifo, (m_player_video.GetDecoderBufferSize()-m_player_video.GetDecoderFreeSpace())>>10, m_player_video.GetDecoderBufferSize()>>10,
                audio_fifo, m_player_audio.GetDelay(), m_player_audio.GetCacheTotal(),
